include/jemalloc/internal/decay.h

```diff
@@ -146,4 +146,14 @@ void decay_reinit(decay_t *decay, nstime_t *cur_time, ssize_t decay_ms);
 bool decay_maybe_advance_epoch(decay_t *decay, nstime_t *new_time,
     size_t current_npages);
 
+typedef struct purge_interval_opts_s purge_interval_opts_t;
+struct purge_interval_opts_s {
+	uint64_t min_interval_ns;
+	uint64_t npages_threshold;
+	uint64_t indefinite_sleep;
```

Comment:
Ditto here; we should just return `(uint64_t)-1` if there are no pages left to purge for the foreseeable future. (This is a theme in jemalloc; we often use -1 to stand in for infinity). It would also be reasonable to pull this into a macro in the header; e.g.

`#define DECAY_UNBOUNDED_TIME_TO_PURGE ((uint64_t)-1)`. The background thread code can see if it gets that value back, and if so convert it to whatever internal representation it likes (I don't think it will need to, but it code).